nengo_viz/static/viz_netgraph_item.js

```diff
@@ -0,0 +1,452 @@
+            }
+            
+            item.set_size(item.size[0] + event.deltaRect.width / w / 2, 
+                          item.size[1] + event.deltaRect.height / h / 2);
+            item.set_position(item.pos[0] + event.deltaRect.width / w / 2 + 
+                                            event.deltaRect.left / w, 
+                              item.pos[1] + event.deltaRect.height / h / 2 + 
+                                            event.deltaRect.top / h);
+            })
+        .on('resizeend', function(event) {
+            var item = ng.svg_objects[uid];
+            item.constrain_position();                
+            ng.notify({act:"pos_size", uid:uid, 
+                       x:item.pos[0], y:item.pos[1],
+                       width:item.size[0], height:item.size[1]});
+            });
+            
+    interact(this.g)
+        .on('wheel', function(event) {
+            var item = ng.svg_objects[uid];
+            var step_size = 1.1; // size of zoom per wheel click
+
+            var delta = event.wheelDeltaY || -event.deltaY
+            var scale = delta > 0 ? step_size : 1.0 / step_size;
+            
+            item.set_size(item.size[0] * scale, item.size[1] * scale);
+            
+            item.constrain_position();                
+            ng.notify({act:"pos_size", uid:uid, 
+                       x:item.pos[0], y:item.pos[1],
+                       width:item.size[0], height:item.size[1]});
+                       
+            event.stopPropagation();           
+            });
+            
+    if (info.type == 'net') {
+        /** tap to expand or collapse a network */
+        interact(this.g)
+            .on('tap', function(event) {
+                ng.toggle_network(uid);
+            });
+
+        /** if a network is flagged to expand on creation, then expand it */
+        if (info.expanded) {
+            this.expand();
+        }
+    }
+
+};
+
+
+/** expand a collapsed network */
+VIZ.NetGraphItem.prototype.expand = function() {
+    this.g.classList.add('expanded');
+    var screen_h = this.get_nested_height() * 
+                   $(this.ng.svg).height() * this.ng.scale;
+
+    /** move the label to the bottom */
+    this.label.setAttribute('transform', 'translate(0, ' + (screen_h) + ')');
+
+    if (!this.expanded) {
+        this.expanded = true;
+        this.ng.g_items.removeChild(this.g);
+        this.ng.g_networks.appendChild(this.g);
+    } else {
+        console.log("expanded a network that was already expanded");
+        console.log(this);
+    }
+    this.ng.notify({act:"expand", uid:this.uid});
+}
+
+
+/** collapse an expanded network */
+VIZ.NetGraphItem.prototype.collapse = function(report_to_server) {
+    this.g.classList.remove('expanded');
+
+    /** move the label back to the middle */
+    this.label.setAttribute('transform', '');
+    
+    /** remove child NetGraphItems and NetGraphConnections */
+    while (this.child_connections.length > 0) {
```

Comment:
The problem with the forEach approach here is that the ```remove``` function is deleting things from the array, so the forEach gets confused and misses half the items.  I could duplicate the array first and then do a forEach on the duplicate, but that seemed messy.